src/utils/domains.ts

```diff
@@ -34,3 +34,20 @@ export const getDomainOptions = (colonyDomains: Domain[]) =>
     })) || [],
     ['value'],
   );
+
+/**
+ * Get a specific domain balance from the array of Colony balances.
+ */
+export const getDomainBalance = (
+  nativeDomainId: number | string,
+  balances?: Colony['balances'] | null,
+) => {
+  const { balance } =
+    balances?.items
+      ?.filter(notNull)
+      .find(
+        ({ domain: { nativeId } }) => nativeId === Number(nativeDomainId),
+      ) || {};
+
+  return balance;
+};
```

Comment:
I'm not sure but could this be potentially merged with `getBalanceForTokenAndDomain` from `utils/tokens`?